### PR TITLE
Add find cells along line

### PR DESCRIPTION
**Pull request by tkoyama010**
### Overview

Add find cells along line

 

resolves https://github.com/pyvista/pyvista/issues/1702

### Details

- None

### CODE REVIEW

File: pyvista/core/dataset.py [hunk 1]
```diff
@@ -1935,6 +1935,49 @@ def find_closest_cell(self, point: Union[int, np.ndarray]) -> Union[int, np.ndar
         closest_cells = np.array([locator.FindCell(node) for node in point])
         return int(closest_cells[0]) if len(closest_cells) == 1 else closest_cells
 
+    def find_cells_along_line(
+        self,
+        pointa: Iterable[float],
+        pointb: Iterable[float],
+        tolerance=0.0,
+    ) -> Union[int, np.ndarray]:
+        """Find index of cells  in this mesh along pointa to pointb line.
```
Comment (akaszynski) [inline pyvista/core/dataset.py:1944]:
```suggestion
        """Find the index of cells in this mesh along a line.

        Line is defined from ``pointa`` to ``pointb``.
```

File: pyvista/core/dataset.py [hunk 2]
```diff
@@ -1935,6 +1935,49 @@ def find_closest_cell(self, point: Union[int, np.ndarray]) -> Union[int, np.ndar
...
+        pointb : iterable(float)
+            Length 3 coordinate of the pointb to query.
+
+        tolerance : float, optional
+            The absolute tolerance to use to find cells along line.
+
+        Returns
+        -------
+        int or numpy.ndarray
+            Index or indices of the cell in this mesh that is closest
```
Comment (akaszynski) [inline pyvista/core/dataset.py:1960]:
```suggestion
            Index or indices of the cell in this mesh that are closest
```

File: pyvista/core/dataset.py [hunk 3]
```diff
@@ -1935,6 +1935,49 @@ def find_closest_cell(self, point: Union[int, np.ndarray]) -> Union[int, np.ndar
...
+        pointa: Iterable[float],
+        pointb: Iterable[float],
+        tolerance=0.0,
+    ) -> Union[int, np.ndarray]:
+        """Find index of cells  in this mesh along pointa to pointb line.
+
+        Parameters
+        ----------
+        pointa : iterable(float)
+            Length 3 coordinate of the pointa to query.
```
Comment (akaszynski) [inline pyvista/core/dataset.py:1949]:
```suggestion
            Length 3 coordinate of the start of the line.
```

File: pyvista/core/dataset.py [hunk 4]
```diff
@@ -1935,6 +1935,49 @@ def find_closest_cell(self, point: Union[int, np.ndarray]) -> Union[int, np.ndar
...
+
+        pointb : iterable(float)
+            Length 3 coordinate of the pointb to query.
```
Comment (akaszynski) [inline pyvista/core/dataset.py:1952]:
```suggestion
            Length 3 coordinate of the end of the line.
```

File: pyvista/core/dataset.py [hunk 5]
```diff
@@ -1935,6 +1935,51 @@ def find_closest_cell(self, point: Union[int, np.ndarray]) -> Union[int, np.ndar
         closest_cells = np.array([locator.FindCell(node) for node in point])
         return int(closest_cells[0]) if len(closest_cells) == 1 else closest_cells
 
+    def find_cells_along_line(
+        self,
+        pointa: Iterable[float],
+        pointb: Iterable[float],
+        tolerance=0.0,
+    ) -> Union[int, np.ndarray]:
```
[the remark this answers is not in the document]
Comment (adeak) [reply pyvista/core/dataset.py:1943]: (If it can, it shouldn't :)

[the remark this answers is not in the document]
Comment (tkoyama010) [reply pyvista/core/dataset.py:1943]:
```suggestion
    ) -> np.ndarray:
```

File: pyvista/core/dataset.py [hunk 6]
```diff
@@ -1935,6 +1935,51 @@ def find_closest_cell(self, point: Union[int, np.ndarray]) -> Union[int, np.ndar
...
+
+        pointb : iterable(float)
+            Length 3 coordinate of the end of the line.
+
+        tolerance : float, optional
+            The absolute tolerance to use to find cells along line.
+
+        Returns
+        -------
+        int or numpy.ndarray
```
[the remark this answers is not in the document]
Comment (tkoyama010) [reply pyvista/core/dataset.py:1961]:
```suggestion
        numpy.ndarray
```

[the remark this answers is not in the document]
Comment (akaszynski) [reply pyvista/core/dataset.py]: Can confirm that this will always return an array.  Thanks for fixing.

**Review by akaszynski — APPROVED** [review]
LGTM.  Thanks!

**Pull request merged**